coral/src/app/features/approvals/acls/AclApprovals.tsx

```diff
@@ -278,31 +276,32 @@ function AclApprovals() {
     },
   ];
 
-  const rows: AclRequestTableRows[] = data.entries.map(
-    ({
-      req_no,
-      acl_ssl,
-      acl_ip,
-      topicname,
-      aclPatternType,
-      environmentName,
-      teamname,
-      topictype,
-      username,
-      requesttimestring,
-    }) => ({
-      id: Number(req_no),
-      acl_ssl: acl_ssl ?? [],
-      acl_ip: acl_ip ?? [],
-      topicname: topicname,
-      prefixed: aclPatternType === "PREFIXED",
-      environmentName: environmentName ?? "-",
-      teamname,
-      topictype,
-      username: username ?? "-",
-      requesttimestring: requesttimestring ?? "-",
-    })
-  );
+  const rows: AclRequestTableRows[] =
+    data?.entries.map(
```

Comment:
I really don't like the need for optional chaining and `|| []`. I think these rows could be computed in the `onSuccess` part of the query, and update a state, so we would not need to dance around like this 🤔 Like:

```ts
const [rows, setRows] = useState<AclRequestTableRows[]>([])

//...

onSuccess: (data) => {
  //...
  const rowsData = data.entries.map(...)
  setRows(rowsData)
  //...
}

//...

<Table rows={rows }... /> // -> should always be at least [], so no TS sadness

Reply:
Hm... yes that would work and it's quite readable!

Reply:
Updated!

- `getRows` function now takes care of it and returns `[]` if `data` is undefined.